doc/examples/reconst_ivim.py

```diff
@@ -0,0 +1,211 @@
+ due to vascular contributions
+
+In the following example we show how to fit the IVIM model
+on a diffusion weighted dataset and visualize the perfusion
+and diffusion. First, we import all relevant modules:
+"""
+
+import matplotlib.pyplot as plt
+
+from dipy.reconst.ivim import IvimModel, ivim_function
+from dipy.data.fetcher import read_ivim
+
+"""
+We get an IVIM dataset using Dipy's data fetcher `read_ivim`.
+This dataset was acquired with 21 b-values in 3 different directions.
+Volumes corresponding to different directions were registered to each
+other, and averaged across directions. Thus, this dataset has 4 dimensions,
+with the length of the last dimension corresponding to the number
+of b-values.
+"""
+
+img, gtab = read_ivim()
+
+"""
+img contains a nibabel Nifti1Image object(with the data)
+and gtab contains a GradientTable object(information about
+the gradients e.g. b-values and b-vectors). We get the
+data from img using `read_data`.
+"""
+
+data = img.get_data()
+print('data.shape (%d, %d, %d, %d)' % data.shape)
+
+"""
+The data has 54 slices, with 256-by-256 voxels in each slice.
+The fourth dimension corresponds to the b-values in the gtab.
+Let us visualize the data by taking a slice midway(z=27) at bvalue = 0.
+"""
+
+z = 33
+b = 20
+
+plt.imshow(data[:, :, z, b].T, origin='lower', cmap='gray')
+plt.axhline(y=100)
+plt.axvline(x=170)
+plt.savefig("ivim_data_slice.png")
+plt.close()
+
+"""
+The marked point in the image shows a section containing cerebral spinal fluid (CSF)
+so it should have a very high f and D*, the area between the right and left is white
+matter so that should be lower, and the region on the right is gray matter and CSF.
+That should give us some contrast to see the values varying across the regions.
+"""
+
+x1, x2 = 160, 180
+y1, y2 = 90, 110
+data_slice = data[x1:x2, y1:y2, z, :]
+
+plt.imshow(data[x1:x2, y1:y2, z, b].T, origin='lower', cmap="gray")
+plt.savefig("CSF_slice.png")
+plt.close()
+
+"""
+Now that we have prepared the datasets we can go forward with
+the ivim fit. Instead of fitting the entire volume, we focus on a
+small section of the slice, to fit the IVIM model. First, we instantiate
+the Ivim model. Here, we use a two-stage approach: first, a tensor
+is fit to the data, and then this tensor as the initial starting point
+for the non-linear fit of IVIM parameters. All initializations are
+passed with the `IvimModel` such as `x0` or `split_b`. If you are using
+Scipy 0.17, you can also set bounds by setting
+`bounds=([0., 0., 0., 0.], [np.inf, 1., 1., 1.]))` while initializing
+the IvimModel. It is recommeded that you upgrade to Scipy 0.17 since
+the fitting results might at times return values which do not make sense
+physically. (Negative f values or D*)
+"""
+
+ivimmodel = IvimModel(gtab, x0=[100., 0.1, 0.01, 0.01])
+
+"""
+To fit the model, call the `fit_method` and pass the data for fitting.
+"""
+
+ivimfit = ivimmodel.fit(data_slice)
```

Comment:
I am getting an error on this line on Python 3 (but not Python 2): 

  ---------------------------------------------------------------------------
  ValueError                                Traceback (most recent call last)
  /Users/arokem/source/dipy/doc/examples/reconst_ivim.py in <module>()
      111 """
      112 
  --> 113 ivimfit = ivimmodel.fit(data_slice)
      114 
      115 """

  /Users/arokem/source/dipy/dipy/reconst/multi_voxel.py in new_fit(self, data, mask)
       31         for ijk in ndindex(data.shape[:-1]):
       32             if mask[ijk]:
  ---> 33                 fit_array[ijk] = single_voxel_fit(self, data[ijk])
       34         return MultiVoxelFit(self, fit_array, mask)
       35     return new_fit

  /Users/arokem/source/dipy/dipy/reconst/ivim.py in fit(self, data, mask)
      195                                       x0_in_mask)
      196         # Use leastsq to get ivim_params
  --> 197         params_in_mask = self._leastsq(data, x0_two_stage)
      198         return IvimFit(self, params_in_mask)
      199 

  /Users/arokem/source/dipy/dipy/reconst/ivim.py in _leastsq(self, data, x0)
      340                                 gtol=gtol,
      341                                 max_nfev=maxfev,
  --> 342                                 args=(bvals, data))
      343             ivim_params = res.x
      344             return ivim_params

  /Users/arokem/anaconda/lib/python3.5/site-packages/scipy/optimize/_lsq/least_squares.py in least_squares(fun, x0, jac, bounds, method, ftol, xtol, gtol, x_scale, loss, f_scale, diff_step, tr_solver, tr_options, jac_sparsity, max_nfev, verbose, args, kwargs)
      755 
      756     if not in_bounds(x0, lb, ub):
  --> 757         raise ValueError("`x0` is infeasible.")
      758 
      759     x_scale = check_x_scale(x_scale, x0)

  ValueError: `x0` is infeasible.